### PR TITLE
Added laz.dll - /myrep - display ship reputations. /laws - Display co…

### PR DESCRIPTION
**Pull request by LazDisco**
…ntraband items in current sector. Expanded help menu, added popup boxes for a wide range of new help commands, contents controlled by cfg. Notification system for when you are carrying illegal cargo into a specific sector. All settings can be managed via cfg file.

### CODE REVIEW

File: Plugins/Public/Laz/Help.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
+// Player Control plugin for FLHookPlugin
+// Feb 2010 by Cannon
```
Comment (dsyalex) [inline Plugins/Public/Laz/Help.cpp]: um

File: Plugins/Public/Laz/Help.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+	// The path to the configuration file.
+	char szCurDir[MAX_PATH];
+	GetCurrentDirectory(sizeof(szCurDir), szCurDir);
+	string scHelpPluginCfgFile = string(szCurDir) + "\\flhook_plugins\\laz_help.cfg";
+}
+
+namespace Help
+{
+	static bool set_helpEnabled;
+	string helpStart;
```
Comment (dsyalex) [inline Plugins/Public/Laz/Help.cpp]: Let's not hardcode the list of possibly help topics.

File: Plugins/Public/Laz/Help.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+			pub::Player::PopUpDialog(iClientID, caption, message, POPUPDIALOG_BUTTONS_CENTER_OK);
+
+			PrintUserCmdText(iClientID, L"OK");
+			return true;
+		}
+		return false;
+	}
+
+	bool Help::UserCmd_HelpStart(uint iClientID, const wstring & wscCmd, const wstring & wscParam, const wchar_t * usage)
```
Comment (dsyalex) [inline Plugins/Public/Laz/Help.cpp]: so much code duplication below here it's crazy

File: Plugins/Public/Laz/Laws.cpp [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+		if (set_lawsJumpEnabled)
+		{
+			uint currentsystem;
+			pub::Player::GetSystem(iClientID, currentsystem);
+
+			for (map<uint, string>::iterator iter = mapHouseAndSystems.begin(); iter != mapHouseAndSystems.end(); iter++)
+			{
+				uint currentSystem = iter->first;
+				string currentHouse = iter->second;
+				if (currentSystem == currentsystem && currentSystem != PrevSystemID)
```
Comment (dsyalex) [inline Plugins/Public/Laz/Laws.cpp]: yeah PrevSystemID could be for a totally different player. I'm assuming you only tested this with one player.
Also, naming two variables 'currentSystem' with different casing?

File: Plugins/Public/Laz/Main.cpp [hunk 5]
```diff
@@ -0,0 +1,279 @@
+// Laz's FLHook Plugin - Template by Alley
+// Current Features:
```
Comment (dsyalex) [inline Plugins/Public/Laz/Main.cpp]: Not going down the "let's make one plugin hold every feature written and completely destroy the concept of the plugin system" route again

File: Plugins/Public/Laz/Main.cpp [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+// Display player's rep to the every faction - /myrep
+// Help.cpp - Display help information from file "laz_help.cfg"
+// Laws.cpp - Display Illegal goods from various factions with the option of informing when someone carries illegal cargo into a specific system.
+// 28th November 2017 - Last Edit
```
Comment (dsyalex) [inline Plugins/Public/Laz/Main.cpp]: This doesn't need to be done with Git

File: Plugins/Public/Laz/Main.cpp [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+		factionReps[wstos(Faction)] = fRep;
+		string FactionVector = wstos(Faction);
+	}
+
+	vector<pair<string, float>> mapVector;
+
+	for (auto iterator = factionReps.begin(); iterator != factionReps.end(); ++iterator) {
+		mapVector.push_back(*iterator);
+	}
+	sort(mapVector.begin(), mapVector.end(), [=](pair<string, float>& a, pair<string, float>& b)
```
Comment (dsyalex) [inline Plugins/Public/Laz/Main.cpp]: I think we got this from Stack Overflow so should probably link to that

File: Plugins/Public/Laz/Main.cpp [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+	{ L"/helpchat", Help::UserCmd_HelpChat, L"Usage: /helpchat" },
+	{ L"/helpchat*", Help::UserCmd_HelpChat, L"Usage: /helpchat" },
+	{ L"/helppower", Help::UserCmd_HelpPower, L"Usage: /helppower" },
+	{ L"/helppower*", Help::UserCmd_HelpPower, L"Usage: /helppower" },
+	{ L"/helpall", Help::UserCmd_HelpAll, L"Usage: /helpAll" },
+	{ L"/helpall*", Help::UserCmd_HelpAll, L"Usage: /helpAll" },
+	{ L"/helprep", Help::UserCmd_HelpRep, L"Usage: /helprep" },
+	{ L"/helprep*", Help::UserCmd_HelpRep, L"Usage: /helprep" },
+//
+	{ L"/laws", Laws::UserCmd_Laws, L"Usage: /laws" },
```
Comment (dsyalex) [inline Plugins/Public/Laz/Main.cpp]: If this is only doing contraband we should give it a different name.

File: Plugins/Public/Laz/Help.cpp (hunk 1 above)
Comment (dsyalex) [inline Plugins/Public/Laz/Help.cpp]: um

File: Plugins/Public/Laz/Help.cpp (hunk 2 above)
Comment (dsyalex) [inline Plugins/Public/Laz/Help.cpp]: Let's not hardcode the list of possible help topics.

File: Plugins/Public/Laz/Help.cpp (hunk 3 above)
Comment (dsyalex) [inline Plugins/Public/Laz/Help.cpp]: When you reach the level of duplication shown below here, it's time to re-think what you should be hardcoding and what you should pull from config.

**Comment by LazDisco** [comment]
Alright. Due to large-scale changes, I'll submit a separate pull request. Don't want a lot of unrelated commits being thrown in the repo.

**Pull request closed without merging**